packages/foundry/contracts/hooks/SwapDiscountHook/DiscountCampaignFactory.sol

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ReentrancyGuard } from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import { IPoolInfo } from "@balancer-labs/v3-interfaces/contracts/pool-utils/IPoolInfo.sol";
+import { TransferHelper } from "./libraries/TransferHelper.sol";
+import { IDiscountCampaignFactory } from "./Interfaces/IDiscountCampaignFactory.sol";
+import { ISwapDiscountHook } from "./Interfaces/ISwapDiscountHook.sol";
+import { IDiscountCampaign } from "./Interfaces/IDiscountCampaign.sol";
+import { DiscountCampaign } from "./DiscountCampaign.sol";
+
+/**
+ * @title DiscountCampaignFactory
+ * @notice Factory contract for creating and managing discount campaigns.
+ * @dev Allows for the creation and updating of discount campaigns associated with liquidity pools.
+ */
+contract DiscountCampaignFactory is ReentrancyGuard, IDiscountCampaignFactory, Ownable {
+    /// @notice Mapping to store discount campaigns associated with pools.
+    mapping(address => CampaignData) public override discountCampaigns;
+
+    /// @notice Address of the SwapDiscountHook contract used to track user swaps.
+    address public swapDiscountHook;
+
+    /**
+     * @notice Initializes the DiscountCampaignFactory contract.
+     * @dev Sets the owner of the contract during deployment.
+     */
+    constructor() Ownable(msg.sender) {}
+
+    /**
+     * @notice Modifier to allow only the campaign owner to execute the function.
+     * @param pool The address of the liquidity pool for which the campaign is managed.
+     */
+    modifier onlyCampaignOwner(address pool) {
+        if (msg.sender != discountCampaigns[pool].owner) revert NOT_AUTHORIZED();
+        _;
+    }
+
+    /**
+     * @notice Modifier to ensure the campaign for a given pool exists.
+     * @param pool The address of the liquidity pool.
+     */
+    modifier campaignExists(address pool) {
+        address campaignAddress = discountCampaigns[pool].campaignAddress;
+        if (campaignAddress == address(0)) revert PoolCampaignDoesnotExist();
+        (, , , , , , address poolAddress, ) = IDiscountCampaign(campaignAddress).campaignDetails();
+        if (pool != poolAddress) {
+            revert PoolAddressCannotBeChanged();
```

Comment:
How could this happen? Should be designed to be impossible, so that you don't need to check for it. (Probably already is; I don't see any way to change the pool address. All you can do is "replace"/"refresh" a campaign with new details, which changes everything at once.)

Reply:
Yes, you are right the second check shouldn't be possible.
Fixed it